### PR TITLE
Added Email Address Enrichment v2.1 + Test v2.1

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
In Progress - need documentation

## Related Issues
fixes: https://github.com/demisto/etc/issues/16830
Related to: https://github.com/demisto/etc/issues/16881

## Description
Added the `Email Address Enrichment - Generic v2.1` playbook. The playbook features the following improvements over the previous one:
- Changed AD to AD v2
- Fixed DomainSquatting filter (3-level nesting filter)
- Fixed Active Directory filter (3-level nesting filter)
- Updated descriptions of task names / playbook
- Fixed playbook domain input bug (used to give a fixed value instead of a reference to the domains)
- Removed redundant outputs
- Improve description of output

Also added the test playbook `Email Address Enrichment - Generic v2.1 - Test`.

## Screenshots
Playbook
image

Test playbook
image

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation
- [ ] Code Review

### CODE REVIEW

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+    timertriggers: []
+    ignoreworker: false
+  "14":
+    id: "14"
+    taskid: e521733d-b152-4a1f-8050-e7658405a187
+    type: condition
+    task:
+      id: e521733d-b152-4a1f-8050-e7658405a187
+      version: -1
+      name: Are there any internal email addresses?
```
Comment (kirbles19) [inline Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: Are there internal email addresses?

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+      description: Checks whether there are email addresses with a NetworkType attribute
```
Comment (kirbles19) [inline Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: Checks whether there is at least one email address with a NetworkType attribute set to "Internal".

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+- contextPath: Account.Email.Distance.Domain
+  description: The compared domain.
+  type: string
+- contextPath: Account.Email.Distance.Value
+  description: 'The distance between the email domain and the compared domain. '
+  type: number
+- contextPath: DBotScore
+  description: The DBotScore object.
+  type: unknown
+releaseNotes: "Fixed a bug where DomainSquatting and Active Directory filters were performing an incorrect check.
```
Comment (kirbles19) [inline Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: The DomainSquatting and Active Directory filters now work as expected.

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+Fixed a bug where the playbook input received a fixed value instead of a reference to a domain value. Replaced Active
```
Comment (kirbles19) [inline Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: Improved playbook inputs.

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+Directory with the Active Directory v2 Query integration. Removed redundant outputs and improved an output description."
```
Comment (kirbles19) [inline Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: Improved output implementation and descriptions.

File: Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml (hunk 3 above)
Comment (idovandijk) [reply Playbooks/playbook-Email_Address_Enrichment_-_Generic_v2.1.yml]: So only the first line is replaced with the new one you wrote

**Pull request merged**